File: ios-app/dbv-nfc-games/Models/Stage.swift
import Foundation

struct Stage: Codable, Identifiable, Hashable {
    let id: UUID
    let gameId: UUID?
    let name: String
    let description: String
    let orderIndex: Int
    let transitionType: String  // "manual", "scheduled", "trigger"
    let scheduledAt: String?
    let triggerBaseId: UUID?
    let isActive: Bool
    let baseIds: [UUID]?
    let createdAt: String?
    let updatedAt: String?
}

struct CreateStageRequest: Encodable {
    let name: String
    let description: String?
    let transitionType: String
    let scheduledAt: String?
    let triggerBaseId: UUID?
}

struct UpdateStageRequest: Encodable {
    let name: String
    let description: String?
    let transitionType: String
    let scheduledAt: String?
    let triggerBaseId: UUID?
}

struct ReorderStagesRequest: Encodable {
    let ids: [UUID]
}

File: web/src/types/stage.ts
export type TransitionType = 'scheduled' | 'trigger' | 'manual'

export interface Stage {
  id: string
  gameId: string
  name: string
  description: string | null
  orderIndex: number
  transitionType: TransitionType
  scheduledAt: string | null
  triggerBaseId: string | null
  isActive: boolean
  baseIds: string[]
  createdAt: string
  updatedAt: string
}
